backend/src/main/java/com/funeat/recipe/application/RecipeService.java

```diff
@@ -0,0 +1,66 @@
+package com.funeat.recipe.application;
+
+import com.funeat.member.domain.Member;
+import com.funeat.member.persistence.MemberRepository;
+import com.funeat.product.domain.ProductRecipe;
+import com.funeat.product.persistence.ProductRecipeRepository;
+import com.funeat.product.persistence.ProductRepository;
+import com.funeat.recipe.domain.Recipe;
+import com.funeat.recipe.domain.RecipeImage;
+import com.funeat.recipe.dto.RecipeCreateRequest;
+import com.funeat.recipe.persistence.RecipeImageRepository;
+import com.funeat.recipe.persistence.RecipeRepository;
+import com.funeat.review.application.ImageService;
+import java.util.List;
+import java.util.Objects;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+@Service
+@Transactional(readOnly = true)
+public class RecipeService {
+
+    private final MemberRepository memberRepository;
+    private final ProductRepository productRepository;
+    private final ProductRecipeRepository productRecipeRepository;
+    private final RecipeRepository recipeRepository;
+    private final RecipeImageRepository recipeImageRepository;
+
+    private final ImageService imageService;
+
+    public RecipeService(final MemberRepository memberRepository, final ProductRepository productRepository,
+                         final ProductRecipeRepository productRecipeRepository, final RecipeRepository recipeRepository,
+                         final RecipeImageRepository recipeImageRepository,
+                         final ImageService imageService) {
+        this.memberRepository = memberRepository;
+        this.productRepository = productRepository;
+        this.productRecipeRepository = productRecipeRepository;
+        this.recipeRepository = recipeRepository;
+        this.recipeImageRepository = recipeImageRepository;
+        this.imageService = imageService;
+    }
+
+    @Transactional
+    public Long create(final Long memberId, final List<MultipartFile> images, final RecipeCreateRequest request) {
+        final Member member = memberRepository.findById(memberId)
+                .orElseThrow(IllegalArgumentException::new);
+
+        final Recipe savedRecipe = recipeRepository.save(new Recipe(request.getName(), request.getContent(), member));
+        request.getProductIds()
+                .stream()
+                .map(it -> productRepository.findById(it)
+                        .orElseThrow(IllegalArgumentException::new))
+                .forEach(it -> productRecipeRepository.save(
+                        new ProductRecipe(it, savedRecipe)
+                ));
```

Comment:
```suggestion
                .forEach(it -> productRecipeRepository.save(new ProductRecipe(it, savedRecipe)));
```

이렇게 붙였을 때 선 넘어가는 거 아니면 붙이는 것도 좋아보여요!